apis/nucleus/src/components/listbox/hooks/useExistingModel.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from 'react';
+import { useModelStore } from '../../../stores/model-store';
+
+export default function useExistingModel({ app, qId, options = {} }) {
+  const [model, setModel] = useState();
+  const [modelStore] = useModelStore();
+  const { sessionModel } = options;
+
+  useEffect(() => {
+    async function fetchObject() {
+      const modelFromStore = modelStore.get(qId);
+      if (modelFromStore) {
+        setModel(modelFromStore);
+      } else {
+        const m = await app.getObject(qId);
+        const key = m ? `${m.id}` : null;
+        modelStore.set(key, m);
```

Comment:
If `key === null` should we still store `m` with key `null`? Throw error instead?

Reply:
I removed the row
`const key = m ? `${m.id}` : null;`

enigma with throw an error if the object isn't found anyway...